KAA-884 Resync of profile manager when meta-flag is set

forGGe opened the pull request:
Profile manager must do resync if corresponding meta-extension flag is sent from the server. 
This branch adds a resync flag to the binary protocol and implement handling of this flag on client side.

forGGe committed:
KAA-884 Get transport channel routine in now public

Couple nitpicks and fixed headers are included in commit as well.

forGGe committed:
KAA-884 Persistance of profile resync flag

There is an unspecified delay between resync request reception
from the server and actual sync of profile body. In different
circumstances enpoint can be switched off before resync action.
Flag must be persisted so that after power-up endpoint will be
aware about last server sync request.

forGGe committed:
KAA-884 Force sync of profile manager and resync check

Allows to trigger a sync process of the profile manager.
If triggered, profile manager will check if full resync is
required.

forGGe committed:
KAA-884 Binary protocol: add resync flag processing

Meta extension part of the protocol is extended in order to force resync
of the profile manager.

forGGe committed:
KAA-884 Tests for profile force sync routine

forGGe committed:
KAA-884 Code review addressed

rasendubi commented on client/client-multi/client-c/src/kaa/utilities/kaa_log.h:
```diff
@@ -29,6 +29,8 @@
 #include "../kaa_error.h"
 #include "../platform/defaults.h"
 
+#include <stdio.h>
```
Kaa core can't require stdio.h

forGGe committed:
KAA-884 Update permanent part of the size of status buffer

New status flag that holds profile resync info affects
a size of a serialization buffer.

forGGe committed:
KAA-884 Update binary encoder test to work with new protocol field

Since meta part of protocol is extended with additional resync field
buffer size and offsets changed as well.

forGGe committed:
KAA-884 Make force sync API call hidden
